plugins/modules/ipaautomember.py

```diff
@@ -0,0 +1,385 @@
+                            grouping, key, inclusiveregex=regex)
+                        commands.append([name, 'automember_add_condition',
+                                         condition_args])
+
+                    for _inclusive in inclusive_del:
+                        key, regex = _inclusive.split("=")
+                        condition_args = gen_condition_args(
+                            grouping, key, inclusiveregex=regex)
+                        commands.append([name, 'automember_remove_condition',
+                                         condition_args])
+
+                if exclusive is not None:
+                    for _exclusive in exclusive_add:
+                        key, regex = _exclusive.split("=")
+                        condition_args = gen_condition_args(
+                            grouping, key, exclusiveregex=regex)
+                        commands.append([name, 'automember_add_condition',
+                                         condition_args])
+
+                    for _exclusive in exclusive_del:
+                        key, regex = _exclusive.split("=")
+                        condition_args = gen_condition_args(
+                            grouping, key, exclusiveregex=regex)
+                        commands.append([name, 'automember_remove_condition',
+                                         condition_args])
+
+            elif state == 'absent':
+                if action == "service":
+                    if res_find is not None:
+                        commands.append([name, 'automember_del',
+                                         {'type': to_text(grouping)}])
+
+                elif action == "member":
+                    if res_find is None:
+                        ansible_module.fail_json(msg="No service '%s'" % name)
+
+                    if inclusive is not None:
+                        for _inclusive in inclusive:
+                            key, regex = _inclusive.split("=")
+                            condition_args = gen_condition_args(
+                                grouping, key, inclusiveregex=regex)
+                            commands.append(
+                                [name, 'automember_remove_condition',
+                                 condition_args])
+
+                    if exclusive is not None:
+                        for _exclusive in exclusive:
+                            key, regex = _exclusive.split("=")
+                            condition_args = gen_condition_args(
+                                grouping, key, exclusiveregex=regex)
+                            commands.append([name,
+                                             'automember_remove_condition',
+                                            condition_args])
+
+            elif state == "rebuild":
+                if res_find is None:
+                    ansible_module.fail_json(msg="No service '%s'" % name)
+                commands.append([None, 'automember_rebuild',
```

Comment:
For `state: rebuilt` the module should accept a list of user and host names.

Reply:
This is the only comment I don't think I've fixed.

To accept a list of objects, what are you proposing? Adding a new module argument just for this?

Reply:
The CLI command automember-rebuild allows to rebuild automember rules for a list of users and/or hosts.

You should add twor parameters to the module `users` and `hosts` that are only valid when `state: rebuilt`. These parameters should be an optional list.